src/Microsoft.AspNetCore.OData/EnableNestedPathsAttribute.cs

```diff
@@ -0,0 +1,150 @@
+    /// applied to the result of the action through query transformations.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public sealed class EnableNestedPathsAttribute : ActionFilterAttribute
+    {
+        /// <inherit/>
+        public EnableNestedPathsAttribute(): base()
+        {
+            // ensures this filter is executed before [EnableQuery] which has default order 0
+            // since the OnActionExecuted method is executed after the action,
+            // a filter with a lower order will be executed after a filter with a higher order
+            Order = 1;
+        }
+
+        /// <summary>
+        /// Transforms the result of the action based on the sequence of property accesses in the odata path
+        /// after the action is executed.
+        /// It first tries to retrieve the IQueryable from the
+        /// returning response message. It then uses the <see cref="ODataPathQueryBuilder"/> to transform
+        /// the query and sets the result back to the response message.
+        /// </summary>
+        /// <param name="actionExecutedContext">The context related to this action, including the response message,
+        /// request message and HttpConfiguration etc.</param>
+        public override void OnActionExecuted(ActionExecutedContext actionExecutedContext)
+        {
+            if (actionExecutedContext == null)
+            {
+                throw Error.ArgumentNull("actionExecutedContext");
+            }
+
+            HttpRequest request = actionExecutedContext.HttpContext.Request;
+            if (request == null)
+            {
+                throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveRequest);
+            }
+
+            ControllerActionDescriptor actionDescriptor = actionExecutedContext.ActionDescriptor as ControllerActionDescriptor;
+            if (actionDescriptor == null)
+            {
+                throw Error.Argument("actionExecutedContext", SRResources.ActionContextMustHaveDescriptor);
+            }
+
+            HttpResponse response = actionExecutedContext.HttpContext.Response;
+
+            ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
+            IQueryable result = responseContent.Value as IQueryable;
+            SingleResult singleResult = responseContent.Value as SingleResult;
+            if (singleResult != null)
+            {
+                // This could be a SingleResult, which has the property Queryable.
+                // But it could be a SingleResult() or SingleResult<T>. Sort by number of parameters
+                // on the property and get the one with the most parameters.
+                PropertyInfo propInfo = responseContent.Value.GetType().GetProperties()
+                    .OrderBy(p => p.GetIndexParameters().Count())
+                    .Where(p => p.Name.Equals("Queryable"))
+                    .LastOrDefault();
```

Comment:
You could use `OrderByDescending` then `FirstOrDefault()`...
I'd assume that `LastOrDefault()` had to perform a count

Reply:
I've also moved the Where filter before the sort.